### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by maxcell**
## Summary

This project had many versions of dependencies to update so this works to make sure everything is up to date as possible. In the process of doing that we are now on Webpack 5 which should afford quite a huge speed boost.

We've also made sure to utilize the NetlifyCMS through the CDN as opposed to a dependency. This follows more to the patterns established on <netlifycms.org>. The package updates also had a huge amount of inspiration from the previous work we did on [victor-hugo from a previous colleague](https://github.com/netlify-templates/victor-hugo/pull/534).

A notable upgrade that we had to do is also replace `node-sass` as it is now through the Dart implementation under `sass`. This at the surface doesn't change any APIs but the internals are significantly different and require the correct bindings for the project.

### Test Plan

For the deploy preview:
- Validate that you can reach all pages in the deploy preview
- Go into the Netlify app and invite yourself as a user to the CMS (https://app.netlify.com/sites/master-template-one-click-hugo-cms/identity). My suggestion would to use something like `+hugo` somewhere in your email before `@` so you can filter it out later if you'd like.
- Accept the invite in your email and set the password
- Navigate to `/admin` under the deploy preview
- Log in with your account
- Make sure you can reach all pages under there
- Make sure to not add a new page or edit because I believe that will automatically commit it to the main branch.

For local development:
- You will need to clone the project down and switch to the different branch
- Install the dependencies
- And make sure you can run with `ntl dev`
- If you'd like to run and validate the CMS functionality, under `static/admin/config.yml` under `backend`, add this line `local_backend: true`
- In one terminal run `npx netlify-cms-proxy-server` and in another run `yarn start`
- This should make it all ready to go!

#### Documentation

- [Webpack 5](https://webpack.js.org)
- [Adding NetlifyCMS to your project documentation](https://www.netlifycms.org/docs/add-to-your-site/)
- [CopyWebpackPlugin Documentation](https://webpack.js.org/plugins/copy-webpack-plugin/#root)
- [`webpack-merge` documentation](https://github.com/survivejs/webpack-merge)
- [OptimizeCSSAssetsWebpackPlugin migration documentation](https://github.com/NMFR/optimize-css-assets-webpack-plugin)
- [Node-sass/lib-sass deprecation](https://sass-lang.com/blog/libsass-is-deprecated)

### CODE REVIEW

**Comment by maxcell** [comment]
You'll notice a TON of files changed and that's because I renamed all the CSS files to SCSS. Especially because many of these files were using Sass it made more sense to rename them. Before, they were just silently failing. Happy to discuss any of these changes more.

**Comment by tzmanics** [comment]
this may just be my machine but whenever i run `yarn start` i get an un-styled version of the site but with `npm start` it is perfect. which is fin except when i run `ntl dev` i only get options for yarn

CleanShot 2022-07-01 at 15 48 40@2x

this could be solved by setting dev settings in the `toml` to `npm start` to match the npm preference in the package.json scripts. if i'm the only one seeing this tho maybe it's not necessary.

**Comment by maxcell** [comment]
Hmmm... I can't seem to reproduce this effect @tzmanics. It could be a timing issue where the build of the HTML finishes first before the CSS is ready and maybe it is a flash unstyled content. Can you try deleting it and trying it again, maybe after installing dependencies let yarn start wait a bit and if nothing happens just refresh the browser page to see if it flashes the content?

**Review by tzmanics — APPROVED** [review]
IT WORKS!!!!!

**Comment by taty2010** [comment]
> If you never see it with `npm start` I'd be down for us changing! One thing I am curious on is if you had any issues installing any dependencies? I had some issues with React 18 so I downgraded locally to React 17 but otherwise is all good!

Ran into this issue when using `npm i` but everything installed just fine with yarn 
Screen Shot 2022-07-06 at 3 15 33 PM

File: site/.hugo_build.lock
Comment (merwok) [inline site/.hugo_build.lock:1]: This file is probably not meant to be committed.

**Pull request merged**